service/common/src/test/java/org/apache/polaris/service/admin/PolarisServiceImplTest.java

```diff
@@ -0,0 +1,239 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.service.admin;
+
+import static org.assertj.core.api.Assertions.assertThatCode;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.when;
+
+import java.lang.reflect.Method;
+import java.util.List;
+import org.apache.polaris.core.PolarisCallContext;
+import org.apache.polaris.core.admin.model.AuthenticationParameters;
+import org.apache.polaris.core.admin.model.Catalog;
+import org.apache.polaris.core.admin.model.CatalogProperties;
+import org.apache.polaris.core.admin.model.ConnectionConfigInfo;
+import org.apache.polaris.core.admin.model.ExternalCatalog;
+import org.apache.polaris.core.admin.model.FileStorageConfigInfo;
+import org.apache.polaris.core.admin.model.PolarisCatalog;
+import org.apache.polaris.core.admin.model.StorageConfigInfo;
+import org.apache.polaris.core.auth.PolarisAuthorizer;
+import org.apache.polaris.core.config.FeatureConfiguration;
+import org.apache.polaris.core.config.PolarisConfigurationStore;
+import org.apache.polaris.core.context.CallContext;
+import org.apache.polaris.core.context.RealmContext;
+import org.apache.polaris.core.persistence.MetaStoreManagerFactory;
+import org.apache.polaris.core.secrets.UserSecretsManagerFactory;
+import org.apache.polaris.service.config.RealmEntityManagerFactory;
+import org.apache.polaris.service.config.ReservedProperties;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+
+public class PolarisServiceImplTest {
+
+  @Mock private RealmEntityManagerFactory entityManagerFactory;
+  @Mock private MetaStoreManagerFactory metaStoreManagerFactory;
+  @Mock private UserSecretsManagerFactory userSecretsManagerFactory;
+  @Mock private PolarisAuthorizer polarisAuthorizer;
+  @Mock private CallContext callContext;
+  @Mock private ReservedProperties reservedProperties;
+  @Mock private PolarisCallContext polarisCallContext;
+  @Mock private PolarisConfigurationStore configurationStore;
+  @Mock private RealmContext realmContext;
+
+  private PolarisServiceImpl polarisService;
+
+  @BeforeEach
+  void setUp() {
+    entityManagerFactory = Mockito.mock(RealmEntityManagerFactory.class);
```

Comment:
Do we need `@Mock` annotations above since we mock explicitly here?

Reply:
Good catch, I have removed it.